src/lightning/fabric/wrappers.py

```diff
@@ -385,3 +387,26 @@ def _capture(model: Any, **kwargs: Any) -> Any:
 
 if _TORCH_GREATER_EQUAL_2_0:
     torch.compile = _capture_compile_kwargs(torch.compile)
+
+
+def _register_backward_hook(tensor: Tensor) -> Tensor:
+    from lightning.fabric.fabric import _in_fabric_backward
+
+    # strategy_requires = is_overridden("backward", self._strategy, parent=Strategy)
+    # precision_requires = any(
+    #     is_overridden(method, self._precision, parent=Precision)
+    #     for method in ("pre_backward", "backward", "post_backward")
+    # )
+    # TODO: How do we get a reference to strategy and precision here?
+    strategy_requires = True
+    precision_requires = True
+
+    def _backward_hook(_) -> None:
+        if (strategy_requires or precision_requires) and not _in_fabric_backward.get():
+            raise RuntimeError(
+                "The current strategy and precision selection requires you to call `fabric.backward(loss)`"
+                " instead of `loss.backward()`."
+            )
+
+    tensor.register_hook(_backward_hook)
```

Comment:
Does the hook get propagated across operations?

```python
y = model(x)
y.register_hook(backward_hook)
z = y * 2
z = y.sum()
z.backward()  # does z have the same backward hook registerered?
```

Reply:
The hook is registered on y, because that's what's returned from the module. It will receive a gradient from (y * 2) and fire.